### PR TITLE
Migration regression/0500/

### PR DESCRIPTION
**Pull request by munificent**
Here's the next batch.

### CODE REVIEW

File: test/tall/regression/0500/0584.unit [hunk 1]
```diff
@@ -2,5 +2,7 @@
 bool
     listEquals<T>(List<T> a, List<T> b) => const ListEquality<T>().equals(a, b);
 <<<
-bool listEquals<T>(List<T> a, List<T> b) =>
-    const ListEquality<T>().equals(a, b);
\ No newline at end of file
+bool listEquals<T>(List<T> a, List<T> b) => const ListEquality<T>().equals(
```
Comment (kallentu) [inline test/tall/regression/0500/0584.unit:5]: Kinda wish this split at the `=>` instead. Especially for the small `a, b` parameters.

**Review by kallentu — APPROVED** [review]

File: test/tall/regression/0500/0584.unit (hunk 1 above)
Comment (munificent) [reply test/tall/regression/0500/0584.unit:5]: Yeah, me too. This is a really tricky corner that I've seen come up in a few places. There are many examples where (to me at least) it's obviously better to split the argument list:

```dart
widget = SomeWidget(
  someParam: value,
  anotherParam: anotherValue,
);
```

But there do sometimes seem to be places like this example where that looks weird and it would be better to split at the `=`/`:`/`=>`. But I've yet to intuit any sort of rule to determine when to split the arguments and when not to.

Maybe it has something to do with the size of the argument list compared to the size of the stuff before the `(`? I'm not sure.

File: test/tall/regression/0500/0506.unit [hunk 2]
```diff
@@ -65,14 +65,15 @@ class Foo {
...
-          null); // async modifier
+        makeIdentifier(param.name),
+        makeParameters(param.parameters),
+        null, // body
+        param.type == null ? null : makeType(param.type),
+        makeEmptyModifiers(), // TODO: Function parameter modifiers?
+        null, // initializers
+        null, // get/set
+        null,
+      ); // async modifier
```
Comment (natebosch) [inline test/tall/regression/0500/0506.unit:76]: I think it's probably fine to expect authors to manually move some edge case comments like this.


Should we move it here too, or leave it as is and let the regression test remain non-representative for code using the new format?

File: test/tall/regression/0500/0584.unit (hunk 1 above)
Comment (natebosch) [reply test/tall/regression/0500/0584.unit:5]: +1

How difficult would it be to have it prefer to split at the `=>` when the expression fits on a single line at that indentation level? Would that heuristic be enough, or would it lead to weird editing experiences?

**Review by natebosch — APPROVED** [review]

File: test/tall/regression/0500/0506.unit (hunk 2 above)
Comment (munificent) [reply test/tall/regression/0500/0506.unit:76]: My rule for regression tests is to always leave them as-is. (That's why, for example, many of them still contain `new` even though it's no longer idiomatic to use it.)
